relay-sampling/src/lib.rs

```diff
@@ -745,23 +749,56 @@ pub struct DynamicSamplingContext {
     /// Set on transaction start, or via `scope.transaction`.
     #[serde(default)]
     pub transaction: Option<String>,
+    /// The sample rate with which this trace was sampled in the SDK. This is a float between 0 and
+    /// 1.
+    #[serde(default, with = "as_string", skip_serializing_if = "Option::is_none")]
+    pub sample_rate: Option<f64>,
     /// The user specific identifier ( e.g. a user segment, or similar created by the SDK
     /// from the user object).
     #[serde(flatten, default)]
     pub user: TraceUserContext,
+    /// Spillover values for backwards/forwards compat
+    #[serde(flatten, default)]
+    pub other: BTreeMap<String, Value>,
 }
 
 impl DynamicSamplingContext {
+    /// Compute the effective sampling rate based on the random "diceroll" and the sample rate from
+    /// the matching rule.
+    pub fn adjust_sample_rate(&self, rule_sample_rate: f64) -> f64 {
```

Comment:
_nit_: This method does not adjust a sample rate, it actually gives you a compensated sample rate back. The active "adjust" suggests that something is mutated. For instance, you could go with `adjusted_sample_rate` or `remaining_sample_rate` or something like that.